src/crossfilter.js

```diff
@@ -43,6 +43,8 @@ function crossfilter() {
 
   var data = [], // the records
       n = 0, // the number of records; data.length
+      filterValue, // the value used for filtering (value, array, function or undefined)
+      filterValuePresent, // true if filterValue contains something
```

Comment:
I think these need to be declared at the dimension level. Each dimension has its own filter, and if I'm not mistaken, declaring them here will result in the values getting overwritten if more than one dimension has an active filter. (As well as unfiltered dimensions returning the filters of filtered dimensions.)

Reply:
Ooops sorry, my bad. Fixed it, with a new test as you suggested.